### PR TITLE
iio: add easy support for buffers

### PR DESCRIPTION
**Pull request by mchindri**
For the moment iio_demo was updated.
Updates to axi and project will come

### CODE REVIEW

**Comment by mchindri** [comment]
v2:
 - Fix cppcheck
 - add commit for ad469x_fmcz

**Review by buha — APPROVED** [review]
lgtm, doesn't seem to break stuff and simplifies buffers

**Comment by mchindri** [comment]
v3:
 - Update tinyiiod submodule. (Some updates where made there and changes on the first commit is for that)
 - Remove commit related to ad469x project, it need more testing
 - apply astyle

**Comment by mchindri** [comment]
v4:
 - astyle again

**Comment by mchindri** [comment]
v5:
 - more astyle

File: libraries/iio/iio.c [hunk 1]
```diff
@@ -698,6 +733,21 @@ static ssize_t iio_transfer_dev_to_mem(const char *device, size_t bytes_count)
...
+	struct iio_data_buffer	*r_buff;
+	uint32_t		samples;
+	ssize_t			ret;
+
+	r_buff = iio_interface->read_buffer;
+	if (r_buff && iio_interface->dev_descriptor->read_dev) {
+		if (bytes_count > r_buff->size)
+			return -ENOMEM;
+		samples = bytes_to_samples(iio_interface, bytes_count);
+		ret = iio_interface->dev_descriptor->read_dev(
```
Comment (cristiantpop) [inline libraries/iio/iio.c:746]: This function looks very similar to transfer_dev_to_mem

Comment (cristiantpop) [reply libraries/iio/iio.c:746]: Maybe we should try and have a single function pointer with diffrent implementations

Comment (cristiantpop) [reply libraries/iio/iio.c:746]: OK, they are removed :)

File: libraries/iio/iio.c [hunk 2]
```diff
@@ -647,7 +647,7 @@ static int32_t iio_open_dev(const char *device, size_t sample_size,
 
 	if (iface->dev_descriptor->prepare_transfer)
 		return iface->dev_descriptor->prepare_transfer(
-				iface->dev_instance, mask);
+			       iface->dev_instance, mask);
```
Comment (cristiantpop) [inline libraries/iio/iio.c:650]: Lots of white space changes...

**Review by cristiantpop — APPROVED** [review]

**Comment by mchindri** [comment]
v6:
 - Move modified space to previous commit to fix astyle

File: libraries/iio/iio.c (hunk 2 above)
Comment (mchindri) [reply libraries/iio/iio.c:650]: Removed

**Review by amiclaus — APPROVED** [review]

**Pull request merged**